contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -605,6 +627,10 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
             );
         }
 
+        // approve the specific amount of WETH required
+        IERC20(WETH).safeApprove(address(positionManager), 0);
+        IERC20(WETH).safeApprove(address(positionManager), _wethBalance);
```

Comment:
Probably better to just do an `approve(wethBalance)` here. If it gets used in the swap (which it should always do), we don't have to reset it everytime

Reply:
I agree that approval should always be 0 after a `swap` or `mint` event I am just concerned that a malfunction in a 3rd party contract (not consuming all the WETH approved in allowance) would break the behaviour of our contract (since approves from non 0 previous approval value [will revert](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v4.0/contracts/token/ERC20/utils/SafeERC20.sol#L40-L42))

Reply:
I feel like we should probably start by avoiding `safeApprove` usage. It has been marked as deprecated for a while now. [`approve` of Base WETH](https://basescan.org/address/0x4200000000000000000000000000000000000006#code) doesn't have that constraint. And directly using `approve` will make our code a bit cleaner

Reply:
Yes, we shouldn't be using safeApprove. My bad on not catching it the first code review.

Reply:
thanks @shahthepro and @DanielVF! Corrected: https://github.com/OriginProtocol/origin-dollar/pull/2266/commits/1321f22abb8223a30288c5d2a6549d4352ec0a69